data_management.py

```diff
@@ -27,4 +28,10 @@ def view_trans_by_date_range(df):
     print(range_date)
     print()
 
-
+def analyze_spending_by_category(df):
+    total_spending = df.groupby('Category')['Amount'].sum()
+    total_spending = total_spending.reset_index()
+    total_spending.columns = ['Category', 'Total Spending']
+    print(f'This is total spending for each category.')
+    print(total_spending)
+    print()
```

Comment:
Can you add an new line to the end of file?

It is a rule for programming.
[FYI]
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline